vpr/src/route/connection_router.cpp

```diff
@@ -477,27 +495,49 @@ void ConnectionRouter<Heap>::timing_driven_expand_neighbour(t_heap* current,
     VTR_LOGV_DEBUG(router_debug_, "      Expanding node %d edge %zu -> %d\n",
                    from_node, size_t(from_edge), to_node_int);
 
-    timing_driven_add_to_heap(cost_params,
-                              current,
-                              from_node,
-                              to_node_int,
-                              from_edge,
-                              target_node);
+    // Check if the node exists in the route tree, needed for RCV code
+
+    bool node_exists = false;
+    if (run_rcv) {
+        node_exists = rcv_path_manager.node_exists_in_tree(current->path_data, 
+                                                            to_node);
+    }
+
+    if (!node_exists || !run_rcv) {
+        timing_driven_add_to_heap(cost_params,
+                                  current,
+                                  from_node,
+                                  to_node_int,
+                                  from_edge,
+                                  target_node,
+                                  run_rcv);
+    }
 }
 
 //Add to_node to the heap, and also add any nodes which are connected by non-configurable edges
 template<typename Heap>
 void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params cost_params,
-                                                       const t_heap* current,
+                                                       t_heap* current,
                                                        const int from_node,
                                                        const int to_node,
                                                        const RREdgeId from_edge,
-                                                       const int target_node) {
+                                                       const int target_node,
+                                                       bool run_rcv) {
     t_heap next;
 
+    // Initalize RCV data struct if needed, otherwise it's set to nullptr
+    if (run_rcv) rcv_path_manager.alloc_path_struct(next.path_data);
```

Comment:
As far as I can tell, the only reason you are calling `alloc_path_struct` here is for `backward_cong` and `backward_delay`, which you copy manually anyways.  Why not just have two locals `float backward_cong`, etc, than messing with dynamic allocation anyways?